src/main/java/bookbob/entity/AppointmentRecord.java

```diff
@@ -0,0 +1,131 @@
+package bookbob.entity;
+
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+//@@author G13nd0n
+public class AppointmentRecord {
+    private List<Appointment> appointments;
+    private DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+
+    //@@author G13nd0n
+    public AppointmentRecord() {
+        this.appointments = new ArrayList<>();
+    }
+
+    //@@author G13nd0n
+    public void addAppointment(Appointment appointment) {
+        appointments.add(appointment);
+        this.sort();
+    }
+
+    //@@author G13nd0n
+    public List<Appointment> findAppointments(String input) {
+        List<Appointment> appointments = new ArrayList<>();
+        String[] inputs = input.split("/");
+        String details = inputs[1];
+        if (inputs[0].equals("n")) {
```

Comment:
Maybe it would be a good idea to do 
String filter = inputs[0];
and then use the filter variable to check the conditions for easier readability.